### PR TITLE
pr-type/bug-fix 7114 - wire up the page size event to alter page size

### LINKED ISSUE #7114: [Feature][config-ui] Connections pagination isn't wired up to page size drop down
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

We have hundreds of repositories and I needed to add ac config scope to them all however you can only select 'all' for the currently displayed page of 10. there is a drop down to select larger page sizes but it does not work.

### What do you expect to happen

I expected the drop down to reload the page with the selected number of items per page

### How to reproduce

have more than 10 repo connection. select the drop down on the bottom right and select any page size other than 10

### Anything else

this is the git diff patch for the fix:
--- a/config-ui/src/routes/connection/connection.tsx
+++ b/config-ui/src/routes/connection/connection.tsx
@@ -53,7 +53,7 @@ export const Connection = () => {
   const [operating, setOperating] = useState(false);
   const [version, setVersion] = useState(1);
   const [page, setPage] = useState(1);
-  const [pageSize] = useState(10);
+  const [pageSize, setPageSize] = useState(10);
   const [scopeId, setScopeId] = useState<ID>();
   const [scopeIds, setScopeIds] = useState<ID[]>([]);
   const [scopeConfigId, setScopeConfigId] = useState<ID>();
@@ -97,7 +97,7 @@ export const Connection = () => {
   const handleHideDialog = () => {
     setType(undefined);
   };
-
+
   const handleShowDeleteDialog = () => {
     setType('deleteConnection');
   };
@@ -332,6 +332,10 @@ export const Connection = () => {
             pageSize,
             total,
             onChange: setPage,
+            onShowSizeChange: (current,size)=> {
+              setPage(1);
+              setPageSize(size);
+            }
           }}
           rowSelection={{
             selectedRowKeys: scopeIds,

### Version

master

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wouldd**
### ⚠️ Pre Checklist

> Please complete _ALL_ items in this checklist, and remove before submitting

- [x] I have read through the [Contributing Documentation](https://devlake.apache.org/community/).
- [x] I have added relevant tests.
- [x] I have added relevant documentation.
- [x] I will add labels to the PR, such as `pr-type/bug-fix`, `pr-type/feature-development`, etc.

### Summary
What does this PR do?
As per the ticket I raised #7114 this just wires up the existing pageSize value to the relevant event handler from the pagination library so that the page does change the number of items shown when a user selects them from the existing drop down box

### Does this close any open issues?
Closes #7114

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: config-ui/src/routes/connection/connection.tsx [hunk 1]
```diff
@@ -332,10 +332,14 @@ export const Connection = () => {
...
             total,
             onChange: setPage,
+            onShowSizeChange: (current,size)=> {
+              setPage(1);
+              setPageSize(size);
+            }
           }}
           rowSelection={{
             selectedRowKeys: scopeIds,
-            onChange: (selectedRowKeys) => setScopeIds(selectedRowKeys as ID[]),
```
Comment (mintsweet) [inline config-ui/src/routes/connection/connection.tsx:338]: Can you rollback this line? You can try executing yarn lint under config-ui, which will prompt you with an error.

**Comment by wouldd** [comment]
regards your comment about it modifying the layout of a page... are you saying that you did not previously see the page dropdown at all? because for me in chrome I always saw the page dropdown. it just did not do anything, hence I considered it a bug.

**Comment by mintsweet** [comment]
@wouldd sorry, I understand that currently there is no operation to switch the page size, but it does not affect page turning. Why do you think this is a issue?

**Comment by wouldd** [comment]
@mintsweet page turning is fine yes. but you can only associated config scopes with teh repos visibile on teh current page. selecting 'all' only selects those on the current page. so when the page size selector doesn't do anything you cannot have any other page size. So in my example where I linked ~800 repositories which I then needed to associate config scopes to. having to do that 10 at a time then manualy change pages is incredibly slow.
potentially it would ultimately be better to allow for a proper means to select all unmapped repos regardless of hte pages. but as quick and easy improvement. just wiring up the already present page size selector to actually change the page size means you can at least do this operation 100 at a time.

**Comment by mintsweet** [comment]
@wouldd I see. Awesome PR, please wait patiently for another commiter's reply.

**Comment by wouldd** [comment]
@mintsweet is there anyone specific that should be tagged for approving?

**Comment by mintsweet** [comment]
@wouldd Sorry, we are preparing for 1.0, I will help you advance this PR, please be patient.

**Comment by Startrekzky** [comment]
Hi @wouldd . Thanks for your contribution. I just reviewed the change and the PR itself is good. However, since the change will cause the inconsistency of the pagination between different pages, I'm not sure if we should merge it.  

Let me know your thoughts on this. Thank you.

**Comment by wouldd** [comment]
@Startrekzky when you say inconsistency, you mean it will only fix it on one page and not all pages? Do you want me to apply the same fix to other places?. It's worth noting that in the majority of instances of you redenering a table it is for things that you are unlikely to have that many of, so the impact of the broken page size is considerably less than here where you can easily have 1000 entries.

**Comment by Startrekzky** [comment]
@wouldd I agree that some lists will have more entries than the others, and your improvement is good. The question was whether we should merge it now or merge it when all similar pages are updated accordingly.

We can merge it now, but I suggest you update the project list's pagination later. Does it make sense?

Cc. @mintsweet

**Review by mintsweet — APPROVED** [review]

**Pull request merged**